src/pages/products/ProductCard.jsx

```diff
@@ -10,10 +10,41 @@ export function ProductCard({
   categoryName,
   rating,
 }) {
+  const { wishlistItems, setWishlistItems } = useWishlistContext()
+  const item = {
+    _id,
+    title,
+    img,
+    price,
+    oldPrice,
+    discount,
+    categoryName,
+    rating,
+  }
+  function addWishlistItem() {
+    for (let i = 0; i < wishlistItems.length; i++) {
+      if (item._id === wishlistItems[i]._id)
+        return setWishlistItems(
+          wishlistItems.filter(product => item._id !== product._id)
+        )
+    }
```

Comment:
Try to use es6 for in or for of loop or array.map.

Reply:
replaced for loop to for of loop